### PR TITLE
[stable/traefik] Specify Annotations for Service Account

### PR DESCRIPTION
**Pull request by lawliet89**
Signed-off-by: Yong Wen Chua <lawliet89@users.noreply.github.com>

#### What this PR does / why we need it:

Support use cases like:

- GKE workload identity: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity
- EKS IAM roles for service accounts: https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by lawliet89** [comment]
@dtomcej any review? I know this is deprecated but the previous PR that introduced a separate service for Prometheus broke Prometheus scraping.

File: stable/traefik/values.yaml [hunk 1]
```diff
@@ -423,6 +423,8 @@ accessLogs:
         # Authorization: redact
 rbac:
   enabled: false
+  serviceAccount:
```
Comment (dtomcej) [inline stable/traefik/values.yaml]: This should be commented out by default please.

File: stable/traefik/templates/rbac.yaml [hunk 2]
```diff
@@ -3,6 +3,10 @@ kind: ServiceAccount
 apiVersion: v1
 metadata:
   name: {{ template "traefik.fullname" . }}
+  {{- with .Values.rbac.serviceAccount.annotations }}
```
Comment (dtomcej) [inline stable/traefik/templates/rbac.yaml:7]: This should be wrapped in an If, so that this is not mandatory

**Review by dtomcej — CHANGES_REQUESTED** [review]

File: stable/traefik/templates/rbac.yaml (hunk 2 above)
Comment (lawliet89) [reply stable/traefik/templates/rbac.yaml]: If annotations are empty, the `with` clause will not execute.

File: stable/traefik/values.yaml (hunk 1 above)
Comment (lawliet89) [reply stable/traefik/values.yaml]: Done

File: stable/traefik/templates/rbac.yaml (hunk 2 above)
Comment (lawliet89) [reply stable/traefik/templates/rbac.yaml:7]: Done

**Comment by lawliet89** [comment]
Not stale

**Comment by lawliet89** [comment]
Not stale

**Comment by lawliet89** [comment]
@dtomcej can I get this approved please?

**Comment by lawliet89** [comment]
Bump.

**Comment by lawliet89** [comment]
Closing. Waste of time keeping this alive.

**Pull request closed without merging**